### PR TITLE
Adding basic folder structure for DFT tool

### PR DESCRIPTION
**Pull request by fgaray**
This is the first step for #2784. Nothing functional yet, this is only the basic folder structure without anything implemented yet.

### CODE REVIEW

**Comment by QuantamHD** [comment]
Awesome! Looks good so far. I'll let @maliberty weigh in.

**Comment by fgaray** [comment]
I fixed the format for OpenRoad.cc but I am not sure why it is failing to build on MacOS

File: src/dft/README.md
Comment (maliberty) [inline src/dft/README.md:5]:
```suggestion
This tool is an implementation of Design For Testing. New nets and logic are added to allow IC designs to
be tested for errors in manufacturing.   Physical imperfections can cause hard failures and variability can cause timing errors.
```

File: src/dft/README.md
Comment (maliberty) [inline src/dft/README.md:11]:
```suggestion
* Scan cells that replace flops with registers that allow for testing.
```

File: src/dft/include/dft/MakeDft.hh
Comment (maliberty) [inline src/dft/include/dft/MakeDft.hh:33]: Use "#pragma once"

File: src/dft/src/CMakeLists.txt
Comment (maliberty) [inline src/dft/src/CMakeLists.txt:75]: Why is this required?  Why is it public?

File: src/OpenRoad.cc
Comment (maliberty) [inline src/OpenRoad.cc:181]: Non-standard but probably better.

**Review by maliberty — CHANGES_REQUESTED** [review]

**Comment by maliberty** [comment]
You'll need to dft add to
https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/docs/contrib/Logger.md
https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/utl/include/utl/Logger.h

**Comment by maliberty** [comment]
Is this just for review?

**Comment by maliberty** [comment]
Mac failure:
```
/Users/openroad/github-action-runners/actions-runner-0/_work/OpenROAD/OpenROAD/build/src/dft/src/CMakeFiles/dft.dir/dftTCL_wrap.cxx:1754:10: fatal error: 'dft/Dft.hh' file not found
2023-01-24T01:23:00.7089340Z #include "dft/Dft.hh"
2023-01-24T01:23:00.7090090Z          ^~~~~~~~~~~~
2023-01-24T01:23:00.8152330Z 1 error generated.
2023-01-24T01:23:00.8261980Z make[2]: *** [src/dft/src/CMakeFiles/dft.dir/CMakeFiles/dft.dir/dftTCL_wrap.cxx.o] Error 1
```

File: src/dft/include/dft/MakeDft.hh
Comment (fgaray) [reply src/dft/include/dft/MakeDft.hh:33]: Yeah, I notice that in the code, but I wonder if it is better to migrate to this instead of using #pragma once?, AFAIK pragma one is not standard.

Comment (maliberty) [reply src/dft/include/dft/MakeDft.hh:33]: No we do not intend to migrate. All modern compilers support it

**Comment by QuantamHD** [comment]
@maliberty Googlers are used to submitting code in small increments. I expect that @fgaray would prefer you that you merged in the boiler plate, and then he could add onto it in a new PR.

It's a very common practice especially for webservers/apis. Typically you would send all the boiler plate in with a stubbed response and submit that. 

In Google our code review tool allowa you to stack PRs on top of each other. GitHub doesn't allow you to do such things which can make these workflows a bit more awkward.

In open source there's typically a higher bar to get code in because contributors can come and go as they please. The idea of small PRs is to reduce the review burden. So it just depends on if that would be helpful or not.

**Comment by maliberty** [comment]
I hate getting one huge PR so I like breaking into steps.  I'm ok so long as there is a clear commitment to reaching a useable milestone.  I think it will slow your progress to have to wait on reviews so there is probably some happy medium in terms of size.

Note https://blog.logrocket.com/using-stacked-pull-requests-in-github/ suggests another workflow that might be closer to what google does.

File: src/dft/src/CMakeLists.txt
Comment (fgaray) [reply src/dft/src/CMakeLists.txt]: It was not required, I removed and moved the CMakeLists.txt to the upper directory

File: src/dft/include/dft/MakeDft.hh
Comment (fgaray) [reply src/dft/include/dft/MakeDft.hh]: That's fair!, I changed the header guards to #pragma once

**Comment by fgaray** [comment]
Thanks @QuantamHD for the context! It makes sense that in an Open Source project we want to accept more atomic PRs. If @maliberty is fine, we can submit this PR and next time I will try to use something like what you suggested (https://blog.logrocket.com/using-stacked-pull-requests-in-github/) so you can see the feature that I want to contribute more complete.

@maliberty I think I fixed and resolved all your comments, so it is ready for review.

File: src/utl/include/utl/Logger.h
Comment (maliberty) [inline src/utl/include/utl/Logger.h:89]: Please put in alphabetical order

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/dft/include/dft/MakeDft.hh
Comment (maliberty) [inline src/dft/include/dft/MakeDft.hh:32]: It is just 
```
#pragma once
```

File: src/utl/include/utl/Logger.h
Comment (rovinski) [reply src/utl/include/utl/Logger.h:89]: PAD needs to move down 3 spots then

Comment (fgaray) [reply src/utl/include/utl/Logger.h]: I sorted all the list

**Review by maliberty — APPROVED** [review]

**Pull request merged**
